tracer/test/Datadog.Trace.Tests/Agent/ApiTests.cs

```diff
@@ -45,6 +45,34 @@ public async Task SendTraceAsync_200OK_AllGood()
             requestMock.Verify(x => x.PostAsync(It.IsAny<ArraySegment<byte>>(), MimeTypes.MsgPack), Times.Once());
         }
 
+        [Theory]
+        [InlineData(429)]
+        [InlineData(413)]
+        [InlineData(408)]
+        public async Task SendTracesAsync_ShouldNotRetry_ForSpecificResponses(int statusCode)
```

Comment:
Nit: Do we have any tests to assert that `SendTracesAsync` retries for other status codes?

Reply:
Good question, there is a test for a `500` that asserts that it retries, it may make sense to just cover some additional codes less than `200` and greater than/equal `300` 

It looks like there are also some error logs in the tests with these retries so I think I have some logic wrong in handling the `413` and `408` so I'll look into those with the tests as well.